src/ducks/apps/components/ApplicationRouting.jsx

```diff
@@ -1,28 +1,41 @@
 import React, { Component } from 'react'
 import { Route, withRouter } from 'react-router-dom'
 
+import { InstallModal } from '../currentAppVersion/Containers'
+
 import ApplicationDetails from './ApplicationDetails'
 import UninstallModal from './UninstallModal'
 
 export class ApplicationRouting extends Component {
   render () {
-    const { installedApps, isFetching, parent, history } = this.props
+    const { apps, installedApps, isFetching, isInstalling, parent, history } = this.props
     return (
       <div>
         <Route path={`/${parent}/:appSlug`} render={({ match }) => {
           if (isFetching) return
-          if (installedApps.length && match.params) {
+          if (installedApps && installedApps.length && match.params) {
             const app = installedApps.find(app => app.slug === match.params.appSlug)
             if (!app) return history.push(`/${parent}`)
```

Comment:
What's happen here if this is an application from the registry (so in `apps` only) but not installed, and there is some installedApps? By reading here, it seems to redirect to the parent. Am I wrong?
The `apps` should content all apps (registry + installedApps), so you can rely on it directly I think.

Reply:
In fact `apps` and `installedApps` are passed by `Discover` and `MyApplications`, respectively, so we can't have both `apps` and `installedApps` in this component.

Reply:
You're right, so what about checking the parent?